modules/local/multiqc.nf

```diff
@@ -0,0 +1,69 @@
+process MULTIQC {
+    label 'process_single'
+
+    conda "bioconda::multiqc=1.15"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/multiqc:1.15--pyhdfd78af_0' :
+        'quay.io/biocontainers/multiqc:1.15--pyhdfd78af_0' }"
+
+    input:
+    path suites_json
+    path multiqc_config
+    path plugin_dir
+
+    output:
+    path "*multiqc_report.html", emit: report
+    path "*_data"              , emit: data
+    path "*_plots"             , optional:true, emit: plots
+    path "versions.yml"        , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    """
+    # Set up local Python environment
+    export PYTHONUSERBASE=\${PWD}/.local
+    export PATH=\${PYTHONUSERBASE}/bin:\${PATH}
+
+    # Install the custom MultiQC plugin first
+    cd ${plugin_dir}
+    pip install --user --no-deps .
+    cd -
```

Comment:
If this is going to be ran often it would make a lot more sense for the container to be published to GHCR that can be ran without needing to run this setup script each and every time. That way it's already baked into a published image that is used instead of:

```
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/multiqc:1.15--pyhdfd78af_0' :
        'quay.io/biocontainers/multiqc:1.15--pyhdfd78af_0' }"
```